File: messageGo/grpc/methodData/service_data.go
package methodData

import "game-message-core/proto"

// 服务信息用于服务 注册和释放
type ServiceDataInput struct {
	MsgVersion  int64             `json:"msgVersion"` // 消息版本号 值为毫秒时间戳
	AppId       string            `json:"appId"`
	ServiceType proto.ServiceType `json:"serviceType"`
	Host        string            `json:"host"`
	Port        int32             `json:"port"`
	MapId       int32             `json:"mapId"`
	Online      int32             `json:"online"`
	MaxOnline   int32             `json:"maxOnline"`
	CreatedAt   int64             `json:"createdAt"`
	UpdatedAt   int64             `json:"updatedAt"`
}

// 服务信息用于服务 注册和释放
type ServiceDataOutput struct {
	MsgVersion int64 `json:"msgVersion"` // 消息版本号 值为毫秒时间戳
	Success    bool  `json:"success"`
}

File: messageDotNet/grpc/MethodData/ServiceData.cs
using System;
using UnityEngine;

/// <summary>
/// 服务信息用于服务 注册和释放 request
/// </summary>
[Serializable]
public class ServiceDataInput
{
    // 消息版本号 值为毫秒时间戳
    public long MsgVersion;
    public string APPID;
    public GameMessageCore.ServiceType ServiceType;
    public string Host;
    public int Port;
    public int MapId;
    public int Online;
    public int MaxOnline;
    public long CreatedAt;
    public long UpdatedAt;


    public string ToJson()
    {
        return JsonUtility.ToJson(this);
    }

}


/// <summary>
/// 服务信息用于服务 注册和释放 response
/// </summary>
[Serializable]
public class ServiceDataOutput
{
    // 消息版本号 值为毫秒时间戳
    public long MsgVersion;
    public bool Success;

    public string ToJson()
    {
        return JsonUtility.ToJson(this);
    }

}